### PR TITLE
Update dependencies and get things compiling/passing CI again.

### PR DESCRIPTION
**Pull request by jsdw**
- Update versions of core libraries.
- Use `scale-value` and `scale-decode` in place of the built-in Value type.
- Fix various things eg new clippy lints.
- Put a bunch of stuff into the workspace Cargo.toml for better reuse, inc all library crates used.

### CODE REVIEW

**Review by tadeohepperle — APPROVED** [review]

File: bin/tx-decoder/Cargo.toml [hunk 1]
```diff
@@ -1,22 +1,30 @@
...
+repository.workspace = true
+description.workspace = true
+edition.workspace = true
+rust-version.workspace = true
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-desub = { path = "../../desub", features = [ "polkadot-js" ] }
+desub = { workspace = true, features = [ "polkadot-js" ] }
```
Comment (niklasad1) [inline bin/tx-decoder/Cargo.toml:15]: yeah I don't like the spaces in features but could be taken care of in another PR

File: desub-legacy/src/substrate_types.rs [hunk 2]
```diff
@@ -47,20 +47,21 @@ pub mod pallet_democracy {
...
 			})
 		}
 	}
 
 	/// Static copy of https://docs.substrate.io/rustdocs/latest/pallet_democracy/enum.Conviction.html
-	#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, RuntimeDebug)]
+	#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, RuntimeDebug, Default)]
 	pub enum Conviction {
 		/// 0.1x votes, unlocked.
+		#[default]
```
Comment (niklasad1) [inline desub-legacy/src/substrate_types.rs:64]: TIL

File: desub-legacy/src/test_suite.rs [hunk 3]
```diff
@@ -25,6 +25,7 @@ pub fn mock_runtime(num: u32) -> RuntimeVersion {
 		spec_version: num,
 		impl_version: num,
 		apis: Cow::from(Vec::new()),
-		transaction_version: 4,
```
Comment (niklasad1) [inline desub-legacy/src/test_suite.rs:28]: Why changed? 

Does scale-value need newer metadata or something?

Comment (jsdw) [reply desub-legacy/src/test_suite.rs:28]: Oh, I just didn't like that it was 4, which is identical to the version of the tx protocol (which is a different version!), so I set it to the version from polkadot iirc

**Review by niklasad1 — APPROVED** [review]

**Review by lexnv — APPROVED** [review]
